### PR TITLE
add messages for versionless good/error cases

### PR DESCRIPTION
**Pull request by rsherget**
- [ ] Considered risk of behavior change or other zero migration impact (https://github.com/OpenLiberty/open-liberty/wiki/Behavior-Changes)
- [ ] Resolved Issues: Fixes #FILLMEIN... (verify `release bug` label if applicable: https://github.com/OpenLiberty/open-liberty/wiki/Open-Liberty-Conventions)
- [ ] Resolved external Known Issues (including APARS): Fixes #FILLMEIN...

### CODE REVIEW

**Review by jimblye — COMMENTED** [review]
50E is duplicated.  
The 2nd one should be 56E

**Comment by rsherget** [comment]
#build

**Comment by wraschke** [comment]
Test comment; please ignore.

File: dev/com.ibm.ws.kernel.feature.core/resources/com/ibm/ws/kernel/feature/internal/resources/ProvisionerMessages.nlsprops [hunk 1]
```diff
@@ -238,30 +238,40 @@ UNKNOWN_PLATFORM_VALUE_ENV_VAR=CWWKF0048E: The {0} value of the PREFERRED_PLATFO
 UNKNOWN_PLATFORM_VALUE_ENV_VAR.explanation=Each value in the platform environment variable must be a known platform. The known platforms are a version of javaee, jakartaee, or microProfile.
 UNKNOWN_PLATFORM_VALUE_ENV_VAR.useraction=Check that the environment variable contains only known platforms.
 
-# CWWKF0049E  {0} = the feature that needs to be specified within PREFERRED_FEATURE_VERSIONS
-UPDATE_MISSING_VERSIONLESS_FEATURE_VAL=CWWKF0049E: The environment variable, PREFERRED_FEATURE_VERSIONS, does not contain a value for {0}
-UPDATE_MISSING_VERSIONLESS_FEATURE_VAL.explanation=For each versionless feature in your configuration, there must be a preferred version value for it within PREFERRED_FEATURE_VERSIONS
-UPDATE_MISSING_VERSIONLESS_FEATURE_VAL.useraction=Add a value in PREFERRED_FEATURE_VERSIONS for the feature
+# CWWKF0049E  {0} = set of duplicate platforms
+DUPLICATE_PLATFORMS=CWWKF0049E: Conflicting platform versions {0}.
```
Comment (dmuelle) [inline dev/com.ibm.ws.kernel.feature.core/resources/com/ibm/ws/kernel/feature/internal/resources/ProvisionerMessages.nlsprops]:
```suggestion
DUPLICATE_PLATFORMS=CWWKF0049E: The following configured platform versions are in conflict: {0}.
```

File: dev/com.ibm.ws.kernel.feature.core/resources/com/ibm/ws/kernel/feature/internal/resources/ProvisionerMessages.nlsprops [hunk 2]
```diff
@@ -238,30 +238,40 @@ UNKNOWN_PLATFORM_VALUE_ENV_VAR=CWWKF0048E: The {0} value of the PREFERRED_PLATFO
...
+UNRESOLVED_VERSIONLESS_FEATURE=CWWKF0050E: The {0} versionless features cannot be resolved
 UNRESOLVED_VERSIONLESS_FEATURE.explanation=Versionless features need a designated platform before they can be resolved.
 UNRESOLVED_VERSIONLESS_FEATURE.useraction=Make sure there is a platform element specified for your feature.
 
-# CWWKF0051I  {0} = the versionless feature: {1} = the feature that was resolved from the versionless feature
-VERSIONLESS_FEATURE_RESOLVED_TO_FEATURE=CWWKF0051I: The {0} versionless feature resolved to {1}
-VERSIONLESS_FEATURE_RESOLVED_TO_FEATURE.explanation=The versioned feature resolved from the versionless feature.
+# CWWKF0051I  {0} = the versionless features: {1} = the features that were resolved from the versionless features
+VERSIONLESS_FEATURE_RESOLVED_TO_FEATURE=CWWKF0051I: The {0} versionless features resolved to {1}
+VERSIONLESS_FEATURE_RESOLVED_TO_FEATURE.explanation=The versioned features resolved from the versionless features.
```
Comment (dmuelle) [inline dev/com.ibm.ws.kernel.feature.core/resources/com/ibm/ws/kernel/feature/internal/resources/ProvisionerMessages.nlsprops]:
```suggestion
VERSIONLESS_FEATURE_RESOLVED_TO_FEATURE.explanation=The versioned features that were resolved from the versionless features.
```

File: dev/com.ibm.ws.kernel.feature.core/resources/com/ibm/ws/kernel/feature/internal/resources/ProvisionerMessages.nlsprops [hunk 3]
```diff
@@ -238,30 +238,40 @@ UNKNOWN_PLATFORM_VALUE_ENV_VAR=CWWKF0048E: The {0} value of the PREFERRED_PLATFO
...
-UNKNOWN_PLATFORM_VERSION=CWWKF0053E: The {0} platform element is not a known platform version
-UNKNOWN_PLATFORM_VERSION.explanation=The platform is known, but the version of the platform does not exist.
-UNKNOWN_PLATFORM_VERSION.useraction=Specify a version of the platform that exists.
+# CWWKF0053I  {0} = the resolved platforms
+RESOLVED_PLATFORM=CWWKF0053I: Feature resolution selected the {0} platforms.
+RESOLVED_PLATFORM.explanation=The platform versions were selected to enable resolution of versionless features. The platform versions were obtained from platform elements, from versioned features, or from an environment variable.
+RESOLVED_PLATFORM.useraction=No action is required.
+
+# CWWKF0054E  {0} = the versionless feature that cannot be resolved: {1} = the resolved platform
+INCOMPATIBLE_VERSIONLESS_FEATURE_WITH_PLATFORM=CWWKF0054E: The {0} versionless feature does not have a version belonging to the {1} platform.
```
Comment (dmuelle) [inline dev/com.ibm.ws.kernel.feature.core/resources/com/ibm/ws/kernel/feature/internal/resources/ProvisionerMessages.nlsprops]:
```suggestion
INCOMPATIBLE_VERSIONLESS_FEATURE_WITH_PLATFORM=CWWKF0054E: The {0} versionless feature does not have a version that belongs to the {1} platform.
```

File: dev/com.ibm.ws.kernel.feature.core/resources/com/ibm/ws/kernel/feature/internal/resources/ProvisionerMessages.nlsprops [hunk 4]
```diff
@@ -238,30 +238,40 @@ UNKNOWN_PLATFORM_VALUE_ENV_VAR=CWWKF0048E: The {0} value of the PREFERRED_PLATFO
...
+INCOMPATIBLE_VERSIONLESS_FEATURE_WITH_PLATFORM.explanation=No version of the feature is available for the platform version. Feature resolution was completed, but did not select a version of the feature.
+INCOMPATIBLE_VERSIONLESS_FEATURE_WITH_PLATFORM.useraction=Either, remove the versionless feature, or specify a different platform version, or use a versioned feature.
```
Comment (dmuelle) [inline dev/com.ibm.ws.kernel.feature.core/resources/com/ibm/ws/kernel/feature/internal/resources/ProvisionerMessages.nlsprops:269]:
```suggestion
INCOMPATIBLE_VERSIONLESS_FEATURE_WITH_PLATFORM.useraction=Either specify a different platform version or remove the versionless feature and use a versioned feature instead.
```

File: dev/com.ibm.ws.kernel.feature.core/resources/com/ibm/ws/kernel/feature/internal/resources/ProvisionerMessages.nlsprops [hunk 5]
```diff
@@ -238,30 +238,40 @@ UNKNOWN_PLATFORM_VALUE_ENV_VAR=CWWKF0048E: The {0} value of the PREFERRED_PLATFO
...
+
+# CWWKF0055E 
+NO_RESOLVED_PLATFORM=CWWKF0055E: Versionless features are being used but there is no platform.
```
Comment (dmuelle) [inline dev/com.ibm.ws.kernel.feature.core/resources/com/ibm/ws/kernel/feature/internal/resources/ProvisionerMessages.nlsprops]:
```suggestion
NO_RESOLVED_PLATFORM=CWWKF0055E: Versionless features are specifed but no platform is configured.
```

File: dev/com.ibm.ws.kernel.feature.core/resources/com/ibm/ws/kernel/feature/internal/resources/ProvisionerMessages.nlsprops [hunk 6]
```diff
@@ -238,30 +238,40 @@ UNKNOWN_PLATFORM_VALUE_ENV_VAR=CWWKF0048E: The {0} value of the PREFERRED_PLATFO
...
+NO_RESOLVED_PLATFORM.explanation=A platform must be determined before versionless features can be resolved. The platform can be determined from a platform element, from a versioned feature, or from an environment variable.
```
Comment (dmuelle) [inline dev/com.ibm.ws.kernel.feature.core/resources/com/ibm/ws/kernel/feature/internal/resources/ProvisionerMessages.nlsprops]:
```suggestion
NO_RESOLVED_PLATFORM.explanation=A platform must be specified before versionless features can be resolved. The platform can be specified in a platform element, from a versioned feature, or in an environment variable.
```

Comment (dmuelle) [reply dev/com.ibm.ws.kernel.feature.core/resources/com/ibm/ws/kernel/feature/internal/resources/ProvisionerMessages.nlsprops]: this is a bit confusing- how is the platform determined from a versioned feature? Do you even need a platform if youre using a versioned feature?

**Review by dmuelle — CHANGES_REQUESTED** [review]
ID review- it's a bit unclear in some of these messages: 
Is using a versioned feature an _alternative_ to specifying a platfoRm and using versionless features?
 or is it _one of the ways_ you can specify a platform that other features will use?

**Comment by rsherget** [comment]
Using a versioned feature is one of the ways a user can specify a platform.

File: dev/com.ibm.ws.kernel.feature.core/resources/com/ibm/ws/kernel/feature/internal/resources/ProvisionerMessages.nlsprops (hunk 6 above)
Comment (rsherget) [reply dev/com.ibm.ws.kernel.feature.core/resources/com/ibm/ws/kernel/feature/internal/resources/ProvisionerMessages.nlsprops]: The feature files contain metadata that tells us what platforms they are a part of, and we use that as one of the ways determine a platform. You do not need a platform if you are just using versioned features, a platform is only needed when using versionless features.

File: dev/com.ibm.ws.kernel.feature.core/resources/com/ibm/ws/kernel/feature/internal/resources/ProvisionerMessages.nlsprops (hunk 4 above)
Comment (dmuelle) [inline dev/com.ibm.ws.kernel.feature.core/resources/com/ibm/ws/kernel/feature/internal/resources/ProvisionerMessages.nlsprops]:
```suggestion
INCOMPATIBLE_VERSIONLESS_FEATURE_WITH_PLATFORM.useraction=Either remove the versionless feature, specify a different platform version, or use a versioned feature.
```

**Comment by rsherget** [comment]
#libby

**Comment by rsherget** [comment]
Build: https://wasrtc.hursley.ibm.com:9443/jazz/web/projects/WS-CD#action=com.ibm.team.build.viewResult&id=_DeodMD-yEe-Hn95lLM_tGQ

**Review by hlhoots — DISMISSED** [review]
LGTM

**Comment by rsherget** [comment]
Build: https://wasrtc.hursley.ibm.com:9443/jazz/web/projects/WS-CD#action=com.ibm.team.build.viewResult&id=_frEKkEBeEe-Hn95lLM_tGQ

**Review by dmuelle — APPROVED** [review]
LGTM, thanks

**Review by hlhoots — APPROVED** [review]
LGTM

**Review by tbitonti — APPROVED** [review]

**Pull request merged**